Userland/Libraries/LibGfx/JPGLoader.cpp

```diff
@@ -1124,30 +1075,26 @@ static bool parse_header(InputMemoryStream& stream, JPGLoadingContext& context)
         case JPG_SOI:
         case JPG_EOI:
             dbgln_if(JPG_DEBUG, "{}: Unexpected marker {:x}!", stream.offset(), marker);
-            return false;
+            return Error::from_string_literal("Unexpected marker");
         case JPG_SOF0:
-            if (!read_start_of_frame(stream, context))
-                return false;
+            TRY(read_start_of_frame(stream, context));
             context.state = JPGLoadingContext::FrameDecoded;
             break;
         case JPG_DQT:
-            if (!read_quantization_table(stream, context))
-                return false;
+            TRY(read_quantization_table(stream, context));
             break;
         case JPG_RST:
-            if (!read_reset_marker(stream, context))
-                return false;
+            TRY(read_reset_marker(stream, context));
             break;
         case JPG_DHT:
-            if (!read_huffman_table(stream, context))
-                return false;
+            TRY(read_huffman_table(stream, context));
             break;
         case JPG_SOS:
             return read_start_of_scan(stream, context);
         default:
-            if (!skip_marker_with_length(stream)) {
+            if (skip_marker_with_length(stream).is_error()) {
                 dbgln_if(JPG_DEBUG, "{}: Error skipping marker: {:x}!", stream.offset(), marker);
-                return false;
+                return Error::from_string_literal("Error skipping marker");
```

Comment:
Same note here about manual error checking / hiding the underlying error. This can also capture the result in a variable and then return `result.release_error()`.